utils/postgis_exampletest.py

```
@@ -0,0 +1,607 @@
+                    screen_body = screen_match.group(1)
+
+            was_forced = tag_has_role(opening, FORCE_ROLE)
+            is_auto_testable = auto_testable_programlisting(body, screen_body)
+            new_opening = remove_role(opening, FORCE_ROLE)
+            if was_forced:
+                if is_auto_testable:
+                    new_opening = remove_role(new_opening, EXTERNAL_STATE_ROLE)
+                else:
+                    new_opening = add_role(new_opening, FORCE_ROLE)
+                    new_opening = remove_role(new_opening, EXTERNAL_STATE_ROLE)
+            elif is_auto_testable and migrating_from_opt_in:
```

Ruff (0.15.20)

[warning] Function definition does not bind loop variable `migrating_from_opt_in`

(B023)